util/dvsim/LintParser.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+r"""Helper class for parsing lint reports into a generic hjson format.
+"""
+import re
+import hjson
+
+from pathlib import Path
+from typing import List, Dict, Tuple
+
+
+class LintParser():
+
+    def __init__(self) -> None:
+        self.buckets = {
+            'warnings': [],
+            'errors': [],
+            'lint_infos': [],
+            'lint_warnings': [],
+            'lint_errors': [],
+            # this bucket is temporary and will be removed at the end of the
+            # parsing pass.
+            'fusesoc-error': []
+        }
+        self.labels = {
+            'warnings': 'Tool Warnings',
+            'errors': 'Tool Errors',
+            'lint_infos': 'Lint Infos',
+            'lint_warnings': 'Lint Warnings',
+            'lint_errors': 'Lint Errors',
+        }
+        self.severities = {
+            'warnings': 'warning',
+            'errors': 'error',
+            'lint_infos': 'info',
+            'lint_warnings': 'warning',
+            'lint_errors': 'error',
+        }
+
+
+    def extract_messages(self, log_content: str, patterns: List[str]) -> None:
+        """
+        This extracts messages from the string buffer log_content.
+        The argument patterns needs to be a list of tuples with
+        (<error_severity>, <pattern_to_match_for>).
+        """
+        for bucket, pattern in patterns:
+            self.buckets[bucket] += \
+                re.findall(pattern, log_content, flags=re.MULTILINE)
+
+    def get_results(self, args: Dict[Path, List[Tuple]]) -> Dict[str, int]:
+        """
+        Parse report and corresponding logfiles and extract error, warning
+        and info messages for each IP present in the result folder
+        """
+
+        log_file_contents = []
+        # Open all log files
+        for path, patterns in args.items():
+            try:
+                with path.open() as f:
+                    log_content = f.read()
+            except IOError as err:
+                self.buckets['errors'] += \
+                    ["IOError: %s" % err]
+            else:
```

Comment:
You could remove `else` (adds extra indentation) and instead add `continue` in `except` clause.

Reply:
Yes the `else` is I believe not really needed here and the call to `extract_messages` can be moved to within the `try` block.